src/main/java/com/rackspacecloud/metrics/ingestionservice/listeners/rawlisteners/deserializer/AvroDeserializer.java

```diff
@@ -31,7 +31,20 @@ public void configure(Map<String, ?> map, boolean b) { }
     public T deserialize(String topicName, byte[] data) {
         if(data == null) return null;
 
+//        String temp = "{\"timestamp\":\"2019-03-19T17:21:02.483Z\",\"accountType\":\"CORE\",\"account\":\"29541\",\"device\":\"818944\",\"deviceLabel\":\"818944-app5.brayleino.co.uk\",\"deviceMetadata\":{},\"monitoringSystem\":\"MAAS\",\"systemMetadata\":{\"checkType\":\"remote.tcp\",\"accountId\":\"acxo8jI3RZ\",\"monitoringZone\":\"mziad\",\"tenantId\":\"hybrid:29541\",\"entityId\":\"enKSnpMCZZ\",\"collectorId\":\"conbQTZpiN\",\"checkId\":\"chnhg75Qhs\"},\"collectionName\":\"tcp\",\"collectionLabel\":\"FTP\",\"collectionTarget\":null,\"collectionMetadata\":{\"event-type-override\":\"ftp\"},\"ivalues\":{\"available\":0},\"fvalues\":{},\"svalues\":{},\"units\":{}}";
+//        data = temp.getBytes();
+
         String stringifiedData = new String(data);
+
+        if(stringifiedData.contains("\"collectionTarget\":null")) {
```

Comment:
You can probably remove these content fixes since the maas-metrics-enrich is now normalizing those fields

https://github.com/Rackspace-Segment-Support/maas-metrics-enrich/blob/master/src/main/java/com/rackspace/umb/maasmetricsenrich/services/EnrichmentService.java#L63-L64

Reply:
Removed it. I also removed `String stringifiedData = new String(data)` and pushed stringification into `debug` and `error` calls. So that stringification only happens in those cases rather than every time which is costly affair.